cms/static/js/views/modals/base_modal.js

```diff
@@ -16,12 +16,26 @@
  *     size of the modal.
  *   viewSpecificClasses: A string of CSS classes to be attached to
  *     the modal window.
- *   addSaveButton: A boolean indicating whether to include a save
+ *   addPrimaryActionButton: A boolean indicating whether to include a primary action
  *     button on the modal.
+ *   primaryActionButtonType: A string to be used as type for primary action button.
+ *   primaryActionButtonTitle: A string to be used as title for primary action button.
  */
 define(['jquery', 'underscore', 'gettext', 'js/views/baseview'],
```

Comment:
**note to reviewers:** focus management code(`tabbable_elements`, `inFocus`, `outFocus`) is taken from  [feedback.js](https://github.com/edx/edx-platform/blob/master/common/static/common/js/components/views/feedback.js)